src/config/configure.in

```diff
@@ -89,6 +89,8 @@ HYPRE_TIME="M4_HYPRE_TIME"
 HYPRE_BUGS="M4_HYPRE_BUGS"
 HYPRE_SRCDIR="M4_HYPRE_SRCDIR"
 
+AC_DEFINE(HYPRE_DEPRECATED(reason), _Pragma(reason), [Deprecated function macro])
```

Comment:
Any idea how to do this with cmake?

Reply:
Hi Rob, I've done it already for CMake here: https://github.com/hypre-space/hypre/blob/4bdd801c6540388bc46909e368da96fccae29d15/src/config/HYPRE_config.h.cmake.in#L174